sockeye/attention.py

```diff
@@ -336,6 +344,117 @@ def attend(att_input: AttentionInput, att_state: AttentionState) -> AttentionSta
         return attend
 
 
+class MultiHeadDotAttention(Attention):
+    """
+    Dot product attention with multiple heads as proposed in Vaswani et al, Attention is all you need.
+    Can be used with a RecurrentDecoder.
+
+    :param input_previous_word: Feed the previous target embedding into the attention mechanism.
+    :param depth: Number of hidden units.
```

Comment:
a comment on terminology: I find the use of "depth" confusing here, as it makes one think of the number of layers, rather than a number of hidden units. I'd rather call it num_hidden or num_hidden_per_head (just a suggestion).